### PR TITLE
feat: change some styles in popover

### PR DESCRIPTION
**Pull request by tanvibhakta**
## Description

Fixes # (issue)

- add box shadow 
- change where scroll is
- update width and height 
- header stays sticky on scroll
- update category tokens for border

Depends on https://github.com/appsmithorg/appsmith/pull/26072/

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Issues raised during DP testing
https://github.com/appsmithorg/design-system/pull/590#issuecomment-1683727274
https://github.com/appsmithorg/design-system/pull/590#issuecomment-1686175670

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by tanvibhakta** [comment]
This PR needs testing to 
- check if the width and max-height conform to zeplin 
- check if the shadow conforms to zeplin

**Comment by shadabbuchh** [comment]
As per the above design file:

1. Box shadow - Pass
2. Max Height - Pass
3. Width - Pass
4. Border - Inconsistent with the borders of other similar components in the same section (Entity Explorer). This is not what we had agreed upon in the technical grooming meeting.
Cc: @rohitagarwal88

File: packages/design-system/src/__theme__/default/index.css [hunk 1]
```diff
@@ -129,13 +130,14 @@
   --ads-v2-colors-content-surface-warning-bg: var(--ads-v2-color-bg-warning);
   --ads-v2-colors-content-surface-special-bg: var(--ads-v2-color-bg-special);
   --ads-v2-colors-content-surface-premium-bg: var(--ads-v2-color-bg-premium);
-  --ads-v2-colors-content-surface-default-border: var(--ads-v2-color-border);
+  --ads-v2-colors-content-surface-default-border: var(--ads-v2-color-border-muted);
```
Comment (tanvibhakta) [inline packages/design-system/src/__theme__/default/index.css:133]: This change impacts 

- popover border
- menu separator
- datepicker border and separator
1. 
Screenshot 2023-08-18 at 5 22 06 PM

File: packages/design-system/src/__theme__/default/index.css [hunk 2]
```diff
@@ -129,13 +130,14 @@
...
   --ads-v2-colors-content-surface-info-border: var(--ads-v2-color-border-info);
   --ads-v2-colors-content-surface-success-border: var(--ads-v2-color-border-success);
   --ads-v2-colors-content-surface-error-border: var(--ads-v2-color-border-error);
   --ads-v2-colors-content-surface-warning-border: var(--ads-v2-color-border-warning);
   --ads-v2-colors-content-surface-special-border: var(--ads-v2-color-border-special);
   --ads-v2-colors-content-surface-premium-border: var(--ads-v2-color-border-premium);
+
   --ads-v2-colors-content-container-default-border: var(--ads-v2-color-border-muted);
```
Comment (tanvibhakta) [inline packages/design-system/src/__theme__/default/index.css:141]: This token can now be deleted and replaced with the above changed token

File: packages/design-system/src/__theme__/default/index.css (hunk 1 above)
Comment (tanvibhakta) [reply packages/design-system/src/__theme__/default/index.css:133]: @albinAppsmith - why is datepicker styled using content tokens? Shouldn't it be using control tokens?

**Comment by tanvibhakta** [comment]
@vasanthappsmith waiting on you to update the designs in zeplin and provide a list of components this has been changed in so that QA can test accurately. Alternatively, please let us know if design thinks it is a better idea to change all the popover borders to gray-300 instead. However, the impact of the latter change is greater and will have to be estimated separately. 
cc @AbhaySM

File: packages/design-system/src/__theme__/default/index.css (hunk 1 above)
Comment (tanvibhakta) [reply packages/design-system/src/__theme__/default/index.css:133]: Clarified with @albinAppsmith over call that only the popover part of date pick is using the content token, the rest is using control.

File: packages/design-system/src/__theme__/default/index.css [hunk 3]
```diff
@@ -129,15 +130,13 @@
...
   --ads-v2-colors-content-surface-premium-bg: var(--ads-v2-color-bg-premium);
-  --ads-v2-colors-content-surface-default-border: var(--ads-v2-color-border);
+  --ads-v2-colors-content-surface-default-border: var(--ads-v2-color-border-muted);
   --ads-v2-colors-content-surface-info-border: var(--ads-v2-color-border-info);
   --ads-v2-colors-content-surface-success-border: var(--ads-v2-color-border-success);
   --ads-v2-colors-content-surface-error-border: var(--ads-v2-color-border-error);
   --ads-v2-colors-content-surface-warning-border: var(--ads-v2-color-border-warning);
   --ads-v2-colors-content-surface-special-border: var(--ads-v2-color-border-special);
   --ads-v2-colors-content-surface-premium-border: var(--ads-v2-color-border-premium);
-  --ads-v2-colors-content-container-default-border: var(--ads-v2-color-border-muted);
```
Comment (tanvibhakta) [inline packages/design-system/src/__theme__/default/index.css:139]: The popover on menu and select is to be testing so that the border exhibits no change.

**Comment by vasanthappsmith** [comment]
Only Menu and Popover fall under the _content_ category. 
Updated Menu and Popover to Border Muted.  

**Menu** : https://zpl.io/xmMY9Zk
**Popover** : https://zpl.io/KGmqBzP

**Comment by shadabbuchh** [comment]
@vasanthappsmith there is a discrepancy between Date picker & Date Range picker border color. Please update the correct color:
Screenshot 2023-08-21 at 5 03 09 PM
Screenshot 2023-08-21 at 5 11 04 PM
Cc: @rohitagarwal88 , @tanvibhakta , @albinAppsmith , @BharghaviK

**Comment by shadabbuchh** [comment]
In storybook tested container border of:

1. Date/Date range picker - _Pass_
Screenshot 2023-08-21 at 5 07 24 PM
Screenshot 2023-08-21 at 5 07 40 PM

2. Menu - _Pass_
Screenshot 2023-08-21 at 5 08 09 PM

3. Popover - _Pass_
Screenshot 2023-08-21 at 5 16 21 PM

4. Select - _Pass_
Screenshot 2023-08-21 at 5 15 57 PM

**Comment by vasanthappsmith** [comment]
@shadabbuchh I made changes to this when we talked about it. I'm not certain why it was saved in its original form.t was saved in its original form.

CleanShot 2023-08-22 at 06 14 54@2x

**Comment by shadabbuchh** [comment]
Verified: Button parent dimensions
Screenshot 2023-08-22 at 10 33 51 AM

**Comment by shadabbuchh** [comment]
Button nesting has been removed
Screenshot 2023-08-22 at 11 55 24 AM

**Review by albinAppsmith — APPROVED** [review]

**Comment by shadabbuchh** [comment]
Verified the latest DP:
Button dimensions are correct now.
Screenshot 2023-08-22 at 1 58 02 PM

**Pull request merged**